core/templates/dev/head/pages/library/library.html

```diff
@@ -43,7 +43,39 @@
           <search-results></search-results>
         </div>
 
-        <div class="oppia-exp-summary-tiles-container" ng-if="!inSearchMode">
+        <div class="oppia-exp-summary-tiles-container" ng-if="inGroupMode" style="margin-bottom: 20px;">
+          <div ng-if="libraryGroups.length == 0"
+               style="font-size: 1.2em; padding-bottom: 40px; padding-top: 40px;">
+              <p style="text-align: center;">
+                <span>Oops, there are no explorations to display.</span>
+              </p>
+          </div>
+
+          <div ng-if="libraryGroups.length > 0"
+               ng-repeat="group in libraryGroups track by $index"
+               style="margin-right: auto; margin-left: auto; max-width: 856px;">
+            <h2 class="oppia-group-page-header" style="font-size: 2em;">
+              <span translate="<[group.header_i18n_id]>"></span>
+            </h2>
+
+            <div ng-repeat="tile in group.activity_summary_dicts" style="display: inline-block;">
+              <exploration-summary-tile ng-if="tile.activity_type == 'exploration'"
+                                        exploration-id="tile.id"
+                                        exploration-title="tile.title"
+                                        last-updated-msec="tile.last_updated_msec"
+                                        objective="tile.objective"
+                                        category="tile.category"
+                                        ratings="tile.ratings"
+                                        num-views="tile.num_views"
+                                        thumbnail-icon-url="tile.thumbnail_icon_url"
+                                        thumbnail-bg-color="tile.thumbnail_bg_color"
+                                        class="protractor-test-exp-summary-tile">
+              </exploration-summary-tile>
+            </div>
+          </div>
+        </div>
+
+        <div class="oppia-exp-summary-tiles-container" ng-if="!inSearchMode && !inGroupMode">
```

Comment:
One thought: instead of having two boolean flags, wouldn't it be better to have a single mode flag that took the values LIBRARY_MODE_GROUP, LIBRARY_MODE_SEARCH, LIBRARY_MODE_INDEX? These constants should be defined both in feconf and in the frontend (the following is the frontend version):

```
    oppia.constant('LIBRARY_PAGE_MODES', {
      GROUP: ...,
      INDEX: ...,
      SEARCH: ...
    })
```

and comments should be added to each to say "make sure this is in sync with the other one".